### PR TITLE
Operator equals

### PR DESCRIPTION
**Pull request by a14n**
## Description

* use `operator ==(Object other)` instead of `operator ==(dynamic other)`. In NNBD you can not override `operator ==(dynamic other)` by `operator ==(Object other)` (that is usually used in framework). Moreover `Object.==` changed its parameter type in 2.9.
* make `operator ==` follow the style guide.

## Related Issues

None

## Tests

None

## Checklist

Before you create this PR confirm that it meets all requirements listed below by checking the relevant checkboxes (`[x]`). This will ensure a smooth and quick review process.

- [x] I read the [contributor guide] and followed the process outlined there for submitting PRs.
- [x] I signed the [CLA].
- [x] I read and followed the [C++, Objective-C, Java style guides] for the engine.
- [x] I read the [tree hygiene] wiki page, which explains my responsibilities.
- [x] I updated/added relevant documentation.
- [x] All existing and new tests are passing.
- [x] I am willing to follow-up on review comments in a timely manner.

## Breaking Change

Did any tests fail when you ran them? Please read [handling breaking changes].

- [x] No, no existing tests failed, so this is *not* a breaking change.
- [ ] Yes, this is a breaking change. *If not, delete the remainder of this section.*
   - [ ] I wrote a design doc: https://flutter.dev/go/template *Replace this with a link to your design doc's short link*
   - [ ] I got input from the developer relations team, specifically from: *Replace with the names of who gave advice*
   - [ ] I wrote a migration guide: https://flutter.dev/go/breaking-changes-template *Replace this with a link to a pull request that adds the migration guide to https://flutter.dev/docs/release/breaking-changes*

[issue database]: https://github.com/flutter/flutter/issues
[contributor guide]: https://github.com/flutter/flutter/wiki/Tree-hygiene#overview
[testing the engine]: https://github.com/flutter/flutter/wiki/Testing-the-engine
[C++, Objective-C, Java style guides]: https://github.com/flutter/engine/blob/master/CONTRIBUTING.md#style
[CLA]: https://cla.developers.google.com/
[tree hygiene]: https://github.com/flutter/flutter/wiki/Tree-hygiene
[handling breaking changes]: https://github.com/flutter/flutter/wiki/Tree-hygiene#handling-breaking-changes

### CODE REVIEW

File: lib/web_ui/lib/src/engine/text/paragraph.dart [hunk 1]
```diff
@@ -666,25 +665,25 @@ class EngineParagraphStyle implements ui.ParagraphStyle {
...
-        _textDirection == typedOther._textDirection ||
-        _fontWeight == typedOther._fontWeight ||
-        _fontStyle == typedOther._fontStyle ||
-        _maxLines == typedOther._maxLines ||
-        _fontFamily == typedOther._fontFamily ||
-        _fontSize == typedOther._fontSize ||
-        _height == typedOther._height ||
-        _textHeightBehavior == typedOther._textHeightBehavior ||
-        _ellipsis == typedOther._ellipsis ||
-        _locale == typedOther._locale;
```
Comment (a14n) [inline lib/web_ui/lib/src/engine/text/paragraph.dart:687]: The old version looked quiet strange with `||` instead of `&&`.

Comment (dnfield) [reply lib/web_ui/lib/src/engine/text/paragraph.dart:687]: This had to be a bug.

**Comment by a14n** [comment]
I don't know why CI is red. Can I merge?

**Comment by dnfield** [comment]
@yjbanov should know.

**Pull request merged**